apps/wallet-mobile/src/Receive/AddressDetail.tsx

```diff
@@ -1,25 +1,322 @@
+  spendingHash?: string
+  title?: string
+}
+
+export function AddressDetail({ address, title, addressDetails }: ShareProps) {
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const SCREEN_WIDTH = Dimensions.get('window').width;
+  const itemsPerPage = 1;
+
+  const data = [
+    { cardType: 'QRCode', title, address },
+    { cardType: 'Details', address, stakingHash: addressDetails?.stakingHash, spendingHash: addressDetails?.spendingHash, title }]
+
+  const totalPages = Math.ceil(data.length / itemsPerPage);
+  const circleIndex = Array.from({ length: totalPages }, (_, index) => index + 1);
+
+  const onPageChange = (event) => {
+    const offset = event.nativeEvent.contentOffset.x;
+    const index = Math.floor(offset / (itemsPerPage * SCREEN_WIDTH - 64));
+    setScrollPosition(index);
+  };
+
+  return (
+    <>
+      <View style={styles.container}>
+        <LinearGradient
+          style={[StyleSheet.absoluteFill, { opacity: 1 }]}
+          start={{ x: 0, y: 0 }}
+          end={{ x: 0, y: 1 }}
+          colors={['#E4E8F7', '#C6F7F7']}
+        />
+
+        <FlatList
+          horizontal
+          showsHorizontalScrollIndicator={false}
+          style={{ height: '100%' }}
+          data={data}
+          keyExtractor={(item, index) => index.toString()}
+          pagingEnabled
+          onScroll={onPageChange}
+          snapToInterval={itemsPerPage * SCREEN_WIDTH}
+          decelerationRate="fast"
+          renderItem={({ item }) => {
+            if (item.cardType === 'QRCode') {
+              return <ShareQRCodeCard title={item.title} address={item.address} />;
+            } else if (item.cardType === 'Details') {
+              return <ShareDetailsCard address={item.address} stakingHash={item.stakingHash} spendingHash={item.spendingHash} title={item.title} />;
+            }
+          }}
+        />
+      </View>
+
+      {
+        totalPages > 1 && (
+          <View style={{ flexDirection: 'row', gap: 6, marginTop: 12 }}>
+            {
+              circleIndex.map((index) => (
+                <View
+                  key={index + 'indexCard'}
+                  style={{
+                    width: 12,
+                    height: 12,
+                    borderRadius: 100,
+                    backgroundColor: (index - 1) === scrollPosition ? colors.buttonBackgroundBlue : COLORS.GRAY,
+                  }}
+                />
+              ))
+            }
+          </View>
+        )
+      }
+    </>
+  )
+}
+
+export function CaptureShareQRCodeCard({ address }: ShareProps) {
+  const LogoRef = useRef();
+
+  return (
+    <View
+      style={[styles.touchableCard]}
+    >
+      <LinearGradient
+        style={[StyleSheet.absoluteFill, { opacity: 1 }]}
+        start={{ x: 0, y: 0 }}
+        end={{ x: 0, y: 1 }}
+        colors={['#E4E8F7', '#C6F7F7']}
+      />
+
+      <Image ref={LogoRef.current} source={require('../assets/img/yoroi-logo-text.png')} style={{ height: 40, width: 40 }} />
```

Comment:
We avoid png images. Could you use the svg and transform it to react native component using this tool: https://transform.tools/svg-to-react-native